### PR TITLE
[MM-9694] Replace event.keyCode deprecated attribute with event.key (with backward compatibility)

### PR DESCRIPTION
**Pull request by gnufede**
#### Summary
event.keyCode is deprecated and doesn't work well with different keyboard layouts. So this PR changes the usage of event.keyCode with event.key leaving backward compatibility to event.keyCode when event.key is not available.

#### Ticket Link
[https://github.com/mattermost/mattermost-server/issues/8359](https://github.com/mattermost/mattermost-server/issues/8359)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Touches critical sections of the codebase (auth, posting, etc.)

### CODE REVIEW

File: stores/message_history_store.jsx [hunk 1]
```diff
@@ -44,12 +44,12 @@ class MessageHistoryStoreClass {
             return null;
         }
 
-        if (keyCode === Constants.KeyCodes.UP) {
+        if (keyCode === Constants.KeyCodes.UP[1]) {
```
Comment (gnufede) [inline stores/message_history_store.jsx]: We would need to change the parameters of this function in order to remove keyCode from it. Couldn't find where the parameters are passed though.

Comment (jespino) [reply stores/message_history_store.jsx]: The message_history_store is a wrapper over the mattermost-redux message history, while all components are migrated to use directly mattermost-redux message_history. I think, now is ready to remove, I have to confirm it.

**Comment by jasonblais** [comment]
Thanks @gnufede! I've pinged our QA team to help test keyboard shortcuts and they're planning to do within the next couple of days.

Appreciate the help!

**Comment by lindy65** [comment]
I tested all shortcut keys on Win 10 on above spinmint and found the following:
1. CTRL+SHIFT+K (direct messages menu) - does nothing
2. CTRL+SHIFT+A (account settings) - does nothing
3. CTRL+SHIFT+M (recent mentions) - brings up what seems like a Chrome browser shortcut menu (see screenshot)

image

4. CTRL+SHIFT+L (set focus to input field) - doesn't work
5. SHIFT+up (reply to last message in a channel) - doesn't work

**Comment by lindalumitchell** [comment]
@jasonblais @gnufede I tested as well (on OSX Chrome), and the ones @lindy65 mentioned above (using `CMD` for Mac rather than the Windows `CTRL`), are all working as expected. So the above issues appear to be Win-specific?

**Comment by jasonblais** [comment]
Thanks guys! @gnufede would you like to help take a look at @lindy65's notes above regarding Windows?

**Comment by gnufede** [comment]
Sure! I'll take a look at @lindy65's notes, I can test it on Windows 8.1 and also on Archlinux. Thank you @lindy65 and @lindalumitchell!

**Comment by gnufede** [comment]
I can confirm that in Archlinux the problem exists as well, both in Firefox and Chrome. The only issue I cannot reproduce is item 5 (Shift+ArrowUp), that works for me in Firefox and Chrome.

The reason for failing shortcuts is that keys are uppercased when Shift is pressed, so the comparison fails. This does not apply to shortcut 5, where ArrowUp sends the same key when Shift is pressed.

I'm adding a small commit that will fix the comparison for items 1-4, and investigate further on item 5, I'll try to reproduce it on Windows 8.1

**Comment by lindy65** [comment]
I have retested all 5 issues above (Win10, Chrome, reddogsofwar, and found no issues. All working as expected.

**Comment by jasonblais** [comment]
Queued for dev review next

**Comment by jespino** [comment]
Thanks @gnufede, I will merge it into master soon, but if you want to take a look to other help wanted ticked or need some help to decide, let me now in pre-release chat :smiley:

**Comment by jespino** [comment]
Merged! Thanks again @gnufede!

**Pull request merged**